src/common/observablevector.ts

```diff
@@ -322,6 +330,11 @@ class ObservableVector<T> extends Vector<T> implements IObservableVector<T> {
    */
   set(index: number, value: T): void {
     let oldValues = [this.at(index)];
+    // Bail if the value does not change.
+    let itemCmp = this._itemCmp;
+    if (itemCmp(oldValues[0], value)) {
```

Comment:
I use `removeFirstOf` which is by-value and `removeFirstWhere` which by-cmp-fn in phosphor 1.0 algos